crates/rolldown/src/types/linking_metadata.rs

```diff
@@ -55,7 +55,7 @@ pub struct LinkingMetadata {
   pub referenced_symbols_by_entry_point_chunk: Vec<SymbolRef>,
 
   /// The dependencies of the module. It means if you want include this module, you need to include these dependencies too.
-  pub dependencies: Vec<ModuleIdx>,
+  pub dependencies: FxHashSet<ModuleIdx>,
```

Comment:
`FxIndexSet` would be better.  Logic related to this field during different build should be consistent, otherwise it'll make the debug harder and may cause producing unstable output/warnings/errors.